fix: allow external secretName instead of a default one

Shaked opened the pull request:
Currently it's not possible to use an external secret source although stated in the documentation: 

> it is possible to override the name of the secret by overriding secretName in the values file. This way an external secret with the keys logzioShippingToken and logzioListener can be placed by other means. 

This commit allows the support of a GitOps approach, where values are pushed into git and should never expose any secrets. 

Ideally this should be covered for all other charts in this repository, otherwise it's not clear how to deploy these charts automatically (GitOps wise at least).

mirii1994 commented on charts/fluentd/templates/secret.yaml:
```diff
@@ -1,9 +1,11 @@
 kind: Secret
 metadata:
-  name: {{ .Values.secretName }}
+  name: logzio-logs-secret
```
Is there a reason this is hard coded and not using `{{ .Values.defaultSecretName }}`?

Shaked replied:
@mirii1994 I missed this one. Looking at the code again, I have made some changes which follow a more common helm values pattern

mirii1994 commented on charts/fluentd/values.yaml:
```diff
@@ -130,7 +130,7 @@ secrets:
   logzioShippingToken: ""
   logzioListener: ""
 
-secretName: logzio-logs-secret
```
Instead of deleting entirely `secretName` from the file, I think we should set is to empty string.

mirii1994 reviewed (changes requested):
Hi @Shaked!
Thank you for your contribution.
Please see my comments.

Shaked commented:
Hey, @mirii1994 I have made some changes that make more sense IMO. Can you look at https://github.com/logzio/logzio-helm/pull/194/commits/f4864a4c70abbdfcec04f3f41b6fcc749d6024f9?

mirii1994 commented on charts/fluentd/templates/_helpers.tpl:
```diff
@@ -74,4 +74,8 @@ Builds the list for exclude paths in the tail for the containers - windows
\ No newline at end of file
+{{- end -}}
+
+{{- define "logzio.secretName" }}
+{{- print .Values.secretName }}
+{{- end -}}
```
This function now does the same thing as using `{{ .Values.secretName }}` directly.

mirii1994 commented on charts/fluentd/values.yaml:
```diff
@@ -130,7 +130,8 @@ secrets:
   logzioListener: ""
 
-secretName: logzio-logs-secret
+defaultSecretName: logzio-logs-secret
```
Now this value is not used anywhere

Shaked replied:
This value has been removed completely https://github.com/Shaked/logzio-helm/commit/f4864a4c70abbdfcec04f3f41b6fcc749d6024f9. I think that GH shows it only because we have a conversation open

mirii1994 commented on charts/fluentd/values.yaml:
```diff
@@ -130,7 +130,8 @@ secrets:
 
-secretName: logzio-logs-secret
+defaultSecretName: logzio-logs-secret
+secretName: ""
```
Now with your changes this should have a default value, and people who want will be able to override it.

mirii1994 replied:
Should have a default value.

Shaked commented:
Hey @mirii1994, I have resolved all the mentioned issues: removed the unused function, removed the usage of the defaultSecretNam, and instead just added `secrets.enabled` condition

mirii1994 commented on charts/fluentd/values.yaml:
```diff
@@ -130,7 +130,8 @@ secrets:
   logzioListener: ""
 
-secretName: logzio-logs-secret
+defaultSecretName: logzio-logs-secret
```
Should be removed.

mirii1994 reviewed (commented):
At the moment, your addition to `secrets.yaml` only allows enabling/disabling the creation of the secret by the helm chart, which could have been done with current version, by setting secretName to empty string.
I'm not sure what is the intention here.
Can you supply more info about your use case?

Shaked commented:
@mirii1994 currently it's not possible to disable the secrets.yaml i.e changing the secretName will not disable the secrets.yaml template: https://github.com/logzio/logzio-helm/blob/master/charts/fluentd/templates/secret.yaml 

My PR supports an option to enable it in case it is needed, otherwise, the secret should be set through an external source. Security wise this solution should be the default behavior IMO.

mirii1994 commented:
@Shaked Got it, thanks for the clarification.
Please remove the variable `defaultSecretName`, and set `secretName` with it's original value.
At the moment we prefer that the default behaviour of the chart will be to manage the secret, so please also add to `values.yaml` the enabled parameter set to true.

Shaked commented:
@mirii1994 done

mirii1994 commented:
@Shaked Thanks for contributing! New version is out! :)
